src/Compilers/CSharp/Portable/Symbols/Symbol_Attributes.cs

```diff
@@ -353,14 +353,7 @@ internal virtual void PostDecodeWellKnownAttributes(ImmutableArray<CSharpAttribu
             if (lazyCustomAttributesBag.SetAttributes(boundAttributes))
             {
                 this.RecordPresenceOfBadAttributes(boundAttributes);
-                if (addToDiagnostics == null)
-                {
-                    this.AddDeclarationDiagnostics(diagnostics);
-                }
-                else
-                {
-                    addToDiagnostics.AddRange(diagnostics);
-                }
+                declarationDiagnostics.AddRange(diagnostics);
```

Comment:
Why don't we override AddDeclarationDiagnostics for the parameter? What is the purpose of making it virtual?